tests/Unit/Attribute/Validator/PathValidatorTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace LastCall\DownloadsPlugin\Tests\Unit\Attribute\Validator;
+
+use LastCall\DownloadsPlugin\Attribute\Validator\PathValidator;
+use LastCall\DownloadsPlugin\Attribute\Validator\ValidatorInterface;
+use LastCall\DownloadsPlugin\Enum\Attribute;
+
+class PathValidatorTest extends AbstractValidatorTestCase
+{
+    private string $attribute = 'path';
+
+    public function testNull(): void
+    {
+        $this->parent->expects($this->once())->method('getName')->willReturn($this->parentName);
+        $this->attributeManager->expects($this->never())->method('get');
+        $this->expectUnexpectedValueException($this->attribute, 'is required');
+        $this->validator->validate(null);
+    }
+
+    public function getInvalidPathTests(): array
+    {
+        return [
+            [true, 'bool'],
+            [false, 'bool'],
+            [123, 'int'],
+            [12.3, 'float'],
+            [['key' => 'value'], 'array'],
+            [(object) ['key' => 'value'], 'stdClass'],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidPathTests
+     */
+    public function testInvalidPath(mixed $invalidPath, string $type): void
+    {
+        $this->parent->expects($this->once())->method('getName')->willReturn($this->parentName);
+        $this->attributeManager->expects($this->never())->method('get');
+        $this->expectUnexpectedValueException($this->attribute, sprintf('must be string, "%s" given', $type));
+        $this->validator->validate($invalidPath);
+    }
+
+    public function getAbsolutePathTests(): array
+    {
+        return [
+            ['C:\Programs\PHP\php.ini'],
+            ['/var/www/project/uploads'],
+        ];
+    }
+
+    /**
+     * @dataProvider getAbsolutePathTests
+     */
+    public function testAbsolutePathFile(string $absolutePath): void
+    {
+        $this->attributeManager->expects($this->once())->method('get')->with(Attribute::VARIABLES)->willReturn([]);
+        $this->parent->expects($this->once())->method('getName')->willReturn($this->parentName);
+        $this->expectUnexpectedValueException($this->attribute, 'must be relative path');
+        $this->validator->validate($absolutePath);
+    }
+
+    public function getOutsidePathTests(): array
+    {
+        return [
+            ['../../other/place', []],
+            ['..../other/place', []],
```

Comment:
I think we should remove all `../`, just collapse them (rather than resolve to absolute path, `str_replace('../', '', $inputPath);` would do); and allow overwriting.

This is all to avoid files escaping this process

Reply:
Currently `..` is not allowed at all.